### ISSUE TITLE
Vive cosmos tracking

### ISSUE THREAD

**Issue by Icedomega13** [LibreVR/Revive#1340]
I didn't expect it to work on this version but just to report, when running revive games on the cosmos I don't get any kind of tracking.  The game launches though so there's a plus.

**Comment by T0X1k75** [comment]
Hi there, yes same here, game start fine but no controller tracking !

**Comment by Armada651** [comment]
Since these are a new type of controller you'll need to provide your own controller binding through the SteamVR Input settings.

**Comment by T0X1k75** [comment]
Its not only the button that is not working, the pointer is not as well, so there is no way to see or point the controller in game ! Its like the Helmet dosent see the controller. But as soon as i clic the Vive icon, the menu comes up and i see the controllers, it exit the menu and they dissapear again !

**Comment by Icedomega13** [comment]
Mapping the controls worked! Thank you!!! Now I just gotta figure out why everything looks so dark in game

**Comment by Armada651** [comment]
@T0X1k75 The controller pose is also mapped in the SteamVR Input settings using Action Poses.

**Comment by Armada651** [comment]
Reopening until default bindings are added.

**Comment by T0X1k75** [comment]
@Armada651, Thanks for your help. When i go in SteamVr Controller Setting i see those (Revive/Oculus) Game i am trying to play. But how can we find wich controller boinding is good for each game? Mine is at Cosmos Legacy Binfings. And i understant that bindingsa will actually have the button working, but if the game dosent even see the controller and i have an error saying '' This game require Oculus toutch controller'' how will button binding solve that ?

**Comment by T0X1k75** [comment]
@Icedomega13, would you explain me what have you binded to have Vive Cosmos controllers work in Revive please?

**Comment by Icedomega13** [comment]
> @Icedomega13, would you explain me what have you binded to have Vive Cosmos controllers work in Revive please?

After launching a game double press the left hand menu button to open steam VR menu. Go to settings and then controller bindings.  From there you select the cosmos controllers. I'll try to publish my binding later tonight.

**Comment by Icedomega13** [comment]
> @Armada651, Thanks for your help. When i go in SteamVr Controller Setting i see those (Revive/Oculus) Game i am trying to play. But how can we find wich controller boinding is good for each game? Mine is at Cosmos Legacy Binfings. And i understant that bindingsa will actually have the button working, but if the game dosent even see the controller and i have an error saying '' This game require Oculus toutch controller'' how will button binding solve that ?

Make sure you are opening the game from within steam vr and not the revive dashboard

**Comment by Margar25** [comment]
I'm totally new to Vive cosmos and I experience the same problem as Icedomega13. Can you tell me Ice if you had to install SteamVR Input System and Unity for binding the controllers? If So, Is there a noob guide how to do it?

**Comment by Icedomega13** [comment]
> I'm totally new to Vive cosmos and I experience the same problem as Icedomega13. Can you tell me Ice if you had to install SteamVR Input System and Unity for binding the controllers? If So, Is there a noob guide how to do it?

After launching a game double press the left hand menu button to open steam VR menu. Go to settings and then controller bindings. From there you select the cosmos controllers. I'll try to publish my binding later tonight.

**Comment by Icedomega13** [comment]
Yo should be able to use the built in controller bindings menu built into the steam vr menu no added software was needed.  (double press the menu button on your left hand controller) Settings>controllers>

**Comment by Margar25** [comment]
Thank you guys. It worked. Now I can fully enjoy Vader Immortal :)

**Comment by Icedomega13** [comment]
> Thank you guys. It worked. Now I can fully enjoy Vader Immortal :)

Great! If you don't mind for me, can you try echo arena? My screen is so black I cant see anything I just want to know if it's just me.  Thanks!

**Comment by T0X1k75** [comment]
@Icedomega13, I dont understand, i just tryed to play the game ''COCO`By Disney'' by starting it from steamVr ... But Still it give me the error that i need a Oculus Touch controller to play the game . I Whent in the SteamVr Menu by th edouble tap on the button on left controller and it was already at VIVE COSMOS CONTROLLER. I have no idear why any of my revive/oculus are seing my controllers. They are not tracking at all !
Any idears ?

**Comment by Icedomega13** [comment]
> @Icedomega13, I dont understand, i just tryed to play the game ''COCO`By Disney'' by starting it from steamVr ... But Still it give me the error that i need a Oculus Touch controller to play the game . I Whent in the SteamVr Menu by th edouble tap on the button on left controller and it was already at VIVE COSMOS CONTROLLER. I have no idear why any of my revive/oculus are seing my controllers. They are not tracking at all !
> Any idears ?

I had to do more than just select the controller but also make a keybind once selected for the game.

**Comment by T0X1k75** [comment]
@Icedomega13, if you could share you keybind it you greatly help me please !

**Comment by Icedomega13** [comment]
> @Icedomega13, if you could share you keybind it you greatly help me please !

Which game? I think you have to make a keybind for each one individually but I may not be certain on that.

**Comment by T0X1k75** [comment]
Actually i tried a lot and all same probleme COC, Spider Man, Vader chapter 1.

**Comment by Icedomega13** [comment]
> Actually i tried a lot and all same probleme COC, Spider Man, Vader chapter 1.

I published my keybind for Echo last night. I would do more but I don't have access to the HMD all the time as it's my friends.  I will try to do some more the next time I am there.

**Comment by Bradgt82** [comment]
I just bought Cosmos and cant get none of these fixes to work.  I didnt even know I needed this or Revive.  I am shocked, mad, disappointed and embarrassed.  I cant enjoy my Oculus library touch controllers not detected.  This is so upsetting.  I should have stayed with gen 1 of oculus headset.  Cosmos is not what I thought.  Vive has no games..  sigh..   Not really sure what to do.. Just made this account to post and seek help.

**Comment by Armada651** [comment]
Cosmos will eventually be supported, if you can't get your own bindings to
work just wait for the next Revive update to provide defaults.

**Comment by Margar25** [comment]
I found it very upsetting as well but once I figured it out how set the
binding, revive etc. I was happy to play Vader Immortal and I hope for
more. I took me like 2 days of everything to actually understand what is
needed to make Cosmo works. Dont give up yet. But definitely I will be more
experienced when it comes to buya next generation vr gear.

**Comment by Icedomega13** [comment]
> I just bought Cosmos and cant get none of these fixes to work. I didnt even know I needed this or Revive. I am shocked, mad, disappointed and embarrassed. I cant enjoy my Oculus library touch controllers not detected. This is so upsetting. I should have stayed with gen 1 of oculus headset. Cosmos is not what I thought. Vive has no games.. sigh.. Not really sure what to do.. Just made this account to post and seek help.

wish I could give ya some confidence but due to the poor tracking in general we returned ours.

**Comment by Bradgt82** [comment]
I have tried all my games through revive and I can't get any bindings to work.  I tried making some myself which is complicated, to say the least, and I tried a public one, only one game out of 20 had one mind you...  I do not understand why anyone would even buy HTC Vice VR over Oculus.  This is just ridiculous.   Everythings updated and works.  I have no issues with tracking. 

I was able to get vader episode one to partially work with someone's public binding but not all features  area working..  I cant use menu systems or anything like that.  I have to hard crash the game just to back out etc.  

I would return this if I hadn't preordered it long ago from HTC and had it shipped, plus its opened and plastic is gone now.  

I have a idea.  I am going to use Oculus gen 1 triple base stations I have and run Oculus through revive, just the sensors for the controls.  Oculus gen 1 controls are the best still....  I will use Cosmos for the headset and Oculus controllers if I can.  Should work..  Ill update if it does.
